queue-jdbc/src/main/java/io/kestra/queue/jdbc/client/JdbcSubscriber.java

```diff
@@ -0,0 +1,111 @@
+package io.kestra.queue.jdbc.client;
+
+import io.kestra.core.exceptions.DeserializationException;
+import io.kestra.core.queues.QueueException;
+import io.kestra.core.utils.Await;
+import io.kestra.core.utils.Either;
+import io.kestra.core.utils.Rethrow;
+import io.kestra.jdbc.runner.JdbcQueueConfiguration;
+import io.kestra.queue.AbstractSubscriber;
+import io.kestra.queue.GenericEvent;
+import io.kestra.queue.QueueSubscriber;
+import io.kestra.queue.QueueUtils;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import static io.kestra.core.utils.Rethrow.throwRunnable;
+
+@Slf4j
+public abstract class JdbcSubscriber<T extends GenericEvent> extends AbstractSubscriber<T> {
+    protected final JdbcQueueClient jdbcQueueClient;
+    protected final String queueName;
+
+    public JdbcSubscriber(
+        Class<T> cls,
+        QueueUtils queueUtils,
+        JdbcQueueClient jdbcQueueClient,
+        String queueName
+    ) {
+        super(cls, queueUtils);
+
+        this.jdbcQueueClient = jdbcQueueClient;
+        this.queueName = queueName;
+    }
+
+    protected abstract Integer pool(JdbcQueueClient.MessageConsumer<String, Exception> messageConsumer);
```

Comment:
```suggestion
    protected abstract Integer poll(JdbcQueueClient.MessageConsumer<String, Exception> messageConsumer);
```

Or I didn't understand what it does ;)